src/base/Solution.cpp

```diff
@@ -0,0 +1,61 @@
+//! @file Solution.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/base/Solution.h"
+#include "cantera/thermo/ThermoPhase.h"
+#include "cantera/kinetics/Kinetics.h"
+#include "cantera/transport/TransportBase.h"
+
+namespace Cantera
+{
+
+Solution::Solution() :
+    m_thermo(nullptr),
+    m_kinetics(nullptr),
+    m_transport(nullptr),
+    m_name("<Solution_name>")
+{}
+
+Solution::Solution(const std::string& infile,
+                   const std::string& phasename) :
+    Solution()
+{
+    // this *may* be a spot to load all pieces of a phase
+    throw NotImplementedError("Solution constructor from file");
+}
+
+std::string Solution::name() const {
+    return m_name;
+}
+
+void Solution::setName(const std::string& name){
+    m_name = name;
+}
+
+std::string Solution::toYAML() const {
+    throw NotImplementedError("Solution::toYAML");
+}
+
+void Solution::setThermoPhase(shared_ptr<ThermoPhase> thermo) {
+    m_thermo = thermo;
+    if (m_thermo) {
```

Comment:
I don’t think so. The `name` of a `Solution` should be assigned directly. The only reason I left a `m_name` with `Phase` is that otherwise all kinds of things would break for those instances when a `ThermoPhase` is instantiated without a `Solution`. I anticipate that this requirement will disappear in the long run.

Reply:
Why should the the `name` be assigned directly rather than coming from an input file? Given that the only way to construct a `Solution` (for now) is from an already-existing `ThermoPhase` object, this would be the opportunity to set at least a default value for that name.

Reply:
From my perspective, `Phase::name` is a duplication of `Phase::id` that is not necessary (see #724, which deprecates usage in Cantera internals). In making this choice, I tried to be consistent with past descriptions, which identified `phase_id` as non-unique and `name` as unique (also: usage of `--id` in ck2cti, `--phase-id` in ck2yaml, and `phase_id` as keyword argument in `Solution`). 

Uniqueness is needed for serialization, which makes only sense for the entire `Solution` object, which is why I moved it there. At the moment, the `name` is overwritten in Python (it's only used within this context); after #724 the `Phase::name` assigned by the parsers will not be used anywhere and can be left empty.

Reply:
The discussion is resolved: `Solution` accesses the name via `Phase`, I.e. there is no distinction any longer.